batch/test/test_batch.py

```diff
@@ -460,3 +460,29 @@ def test_create_idempotence(self):
         b = builder._create(batch_token=batch_token)
         b2 = builder._create(batch_token=batch_token)
         assert b.id == b2.id
+
+    def test_batch_create_validation(self):
+        bad_configs = [
+            # unexpected field fleep
+            {'billing_project': 'foo', 'n_jobs': 5, 'token': 'baz', 'fleep': 'quam'},
+            # billing project None/missing
+            {'billing_project': None, 'n_jobs': 5, 'token': 'baz'},
+            {'n_jobs': 5, 'token': 'baz'},
+            # n_jobs None/missing
+            {'billing_project': 'foo', 'n_jobs': None, 'token': 'baz'},
+            {'billing_project': 'foo', 'token': 'baz'},
+            # n_jobs wrong type
+            {'billing_project': 'foo', 'n_jobs': '5', 'token': 'baz'},
+            # token None/missing
+            {'billing_project': 'foo', 'n_jobs': 5, 'token': None},
+            {'billing_project': 'foo', 'n_jobs': 5},
+            # callback None
+            {'billing_project': 'foo', 'callback': None, 'n_jobs': 5, 'token': 'baz'},
```

Comment:
Don't we want to allow callback to be None? What if there's no callback?

Reply:
I was thinking no callback URL would be represented as no callback field.  I guess we could treat a null callback in the same way.  This is annoying and I don't think we're super consistent about this overall.  I'll fix, permissive on input seems better.